rust/worker/src/execution/utils.rs

```diff
@@ -0,0 +1,36 @@
+use std::cmp::Ordering;
+
+#[derive(Clone, Debug)]
+pub struct RecordDistance {
+    pub offset_id: u32,
+    pub measure: f32,
+}
+
+impl PartialEq for RecordDistance {
+    fn eq(&self, other: &Self) -> bool {
+        self.measure.eq(&other.measure)
+    }
+}
+
+impl Eq for RecordDistance {}
+
+impl Ord for RecordDistance {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.measure.total_cmp(&other.measure)
+    }
+}
+
+impl PartialOrd for RecordDistance {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+pub fn normalize(vector: &[f32]) -> Vec<f32> {
```

Comment:
i think we should move normalize to distance crate

Reply:
Update: this is moved to distance crate.